src/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -569,11 +583,21 @@ internal readonly struct TestAccessor
         public TestAccessor(LspWorkspaceManager manager)
             => _manager = manager;
 
-        public Workspace? GetLspMiscellaneousFilesWorkspace()
+        public ValueTask<bool> IsMiscellaneousFilesDocumentAsync(TextDocument document)
+        {
+            return _manager._lspMiscellaneousFilesWorkspaceProvider!.IsMiscellaneousFilesDocumentAsync(document, CancellationToken.None);
+        }
+
+        public async IAsyncEnumerable<T> GetMiscellaneousDocumentsAsync<T>(Func<Project, IEnumerable<T>> documentSelector) where T : TextDocument
         {
-            // For purposes of testing, we test against the implementation that is also a Workspace.
-            // TODO: once we also test the FileBasedPrograms implementation, we need to do something else here.
-            return _manager._lspMiscellaneousFilesWorkspaceProvider as Workspace;
+            foreach (var workspace in _manager._lspWorkspaceRegistrationService.GetAllRegistrations())
+            {
+                foreach (var document in workspace.CurrentSolution.Projects.SelectMany(documentSelector))
```

Comment:
yeah - I think this logic should really be in the provider.  I don't think the queue should necessarily know about how exactly the misc provider is configured